app/src/main/java/homer/model/thermometer/SimpleThermometer.java

```diff
@@ -0,0 +1,46 @@
+package homer.model.thermometer;
+
+import java.util.Objects;
+
+import homer.DeviceInfoImpl;
+import homer.api.Device;
+import homer.api.DeviceIdImpl;
+import homer.api.DeviceInfo;
+import homer.common.Temperature;
+import homer.model.environment.Environment;
+import homer.model.environment.EnvironmentObject;
+
+/**
+ * Implementation of a {@link Device} which returns the temperature of the
+ * environment as it is, without errors or lags.
+ */
+public final class SimpleThermometer implements Device<Temperature>, EnvironmentObject {
+
+    private static final String DEVICE_TYPE = "Thermometer";
+    private Environment environment;
+
+    /**
+     * Creates a new {@link SimpleThermometer}.
+     * 
+     * @param environment The environment in which the thermometer is placed.
+     */
+    public SimpleThermometer(final Environment environment) {
+        this.environment = Objects.requireNonNull(environment);
+    }
+
+    @Override
+    public DeviceInfo getInfo() {
+        return new DeviceInfoImpl(new DeviceIdImpl(), DEVICE_TYPE);
```

Comment:
Not sure about the device creating the id, maybe it should have been the controller creating the id.
Also, should we check for the id to not already exist?

Reply:
Yep, I think the controller has to create a DeviceInfo object and pass it to the Device constructor.

I'm not sure about the check, need to check UUID docs, maybe there's something already implemented for this purpose.

Reply:
I was thinking the controller creating an id and storing the device in a map id -> device, but with the type being reported by the device.

The probability of a collision with the random UUID is extremely unlikely. Anyway if the controller is creating the id, it would be easy to check and regenerate another one.

Reply:
[I don't think we need to worry about collisions](https://stackoverflow.com/questions/24876188/how-big-is-the-chance-to-get-a-java-uuid-randomuuid-collision)